synthesizer/process/src/verify_execution.rs

```diff
@@ -131,6 +131,14 @@ impl<N: Network> Process<N> {
         let num_instances = verifier_inputs.values().map(|(_, inputs)| inputs.len()).sum::<usize>();
         // Ensure the number of instances matches the number of transitions.
         ensure!(num_instances == execution.transitions().len(), "The number of verifier instances is incorrect");
+        // Ensure the same signer is used for all transitions.
+        if let Some(first_transition) = execution.transitions().next() {
```

Comment:
Any cleaner way to do this? Maybe a `try_fold` or some sort of `.all` call without having to do the `.next()`

Reply:
7ec8642b5